receiver/nvmlreceiver/client.go

```diff
@@ -278,6 +319,99 @@ func (client *nvmlClient) collectDeviceMemoryInfo() []nvmlMetric {
 	return deviceMetrics
 }
 
+func (client *nvmlClient) collectProcessMetrics() []processMetric {
+	if client.disable || !client.collectProcessInfo {
+		return nil
+	}
+
+	processMetrics := make([]processMetric, 0)
+	for gpuIndex, device := range client.devices {
+		pids, ret := nvmlDeviceGetAccountingPids(device)
+		if ret != nvml.SUCCESS {
+			msg := fmt.Sprintf("Unable to query cached PIDs on '%v", nvml.ErrorString(ret))
+			client.issueWarningForFailedQueryUptoThreshold(gpuIndex, "nvml.processes", msg)
+			continue
+		}
+
+		for _, pid := range pids {
+			metricName := fmt.Sprintf("nvml.processes{pid=%d}", pid)
+
+			stats, ret := nvml.DeviceGetAccountingStats(device, uint32(pid))
+			if ret != nvml.SUCCESS {
+				msg := fmt.Sprintf("Unable to query pid %d account statistics on '%v", pid, nvml.ErrorString(ret))
+				client.issueWarningForFailedQueryUptoThreshold(gpuIndex, metricName, msg)
+				continue
+			}
+
+			if stats.IsRunning != 1 {
+				continue
+			}
+
+			metric := processMetric{
+				time:                   time.Now(),
+				processPid:             pid,
+				gpuIndex:               uint(gpuIndex),
+				runningTime:            time.Since(time.UnixMicro(int64(stats.StartTime))),
+				lifetimeGpuUtilization: uint64(stats.GpuUtilization),
+				lifetimeGpuMaxMemory:   stats.MaxMemoryUsage,
```

Comment:
Since it's possible to have some devices in MIG mode and some not, I changed the behavior to not disable process collection unless no devices support it.